### PR TITLE
Move to new Zephyr SDK 0.9.5

### PR DESCRIPTION
**Pull request by nashif**
This integrates new SDK and replaces qemu_cortex_m3 with mps2_an385 and
qemu_riscv32 with hifive1.

### CODE REVIEW

**Review by dbkinder — APPROVED** [review]
LGTM

**Comment by dbkinder** [comment]
does this imply a new SDK release is in the works?

**Comment by jukkar** [comment]
I have been investigating how to use e1000 ethernet support in qemu. The current 0.9.3 sdk is missing some options and rom files so I could not get this to work with the existing qemu. Is this new sdk somewhere so I could try it and possibly figure out what options are needed for the e1000 driver to work properly?

**Comment by pfalcon** [comment]
@jukkar : Sure, https://github.com/zephyrproject-rtos/meta-zephyr-sdk/releases

File: arch/arm/include/cortex_m/exc.h [hunk 1]
```diff
@@ -68,15 +68,13 @@ static ALWAYS_INLINE bool _IsInIsr(void)
 		/* If not in thread mode, and if RETTOBASE bit in ICSR is 0,
 		 * then there are preempted active exceptions to execute.
 		 */
-#ifndef CONFIG_BOARD_QEMU_CORTEX_M3
 		/* The polarity of RETTOBASE is incorrectly flipped in
 		 * all but the very latest master tip of QEMU's NVIC driver,
 		 * see commit "armv7m: Rewrite NVIC to not use any GIC code".
 		 * Until QEMU 2.9 is released, and the SDK is updated to
 		 * include it, skip this check in QEMU.
```
Comment (andrewboie) [inline arch/arm/include/cortex_m/exc.h]: Get rid of this comment as well as the ifdef

File: samples/bluetooth/beacon/sample.yaml [hunk 2]
```diff
@@ -4,5 +4,5 @@ sample:
 tests:
   test:
     harness: bluetooth
-    platform_whitelist: qemu_cortex_m3 qemu_x86 nrf52_pca10040
```
Comment (andrewboie) [inline samples/bluetooth/beacon/sample.yaml]: concerned about these changes to the YAML files.
shouldn't the new name be on these whitelists?
else the test just won't be run under emulation unless I'm missing something

**Review by tyrantlizard77 — DISMISSED** [review]

**Comment by jukkar** [comment]
I wonder if we could add necessary ROM files to the SDK so that e1000 driver would work with the SDK. Currently one needs to use host QEMU installed ROM files when starting the qemu if e1000 needs to be used. Unfortunately atm I have no list of the ROM files that are needed for e1000, but I can try to figure it out.

**Comment by jukkar** [comment]
After copying these files from host qemu to sdk qemu, the zephyr boots with e1000 driver. The e1000 still does not work properly after that but at least the device boots ok.
```
bios-256k.bin
efi-e1000.rom
pxe-e1000.rom
```

File: samples/bluetooth/beacon/sample.yaml (hunk 2 above)
Comment (nashif) [reply samples/bluetooth/beacon/sample.yaml]: it will run under qemu_x86

**Review by nategraff-sifive — APPROVED** [review]
qemu_riscv32 and hifive1 changes LGTM

**Comment by nashif** [comment]
@jukkar 0.9.5 now has the roms

**Comment by galak** [comment]
I think you'll need the fixes in #11059 & #11061 for this to work/make progress.

File: boards/riscv32/qemu_riscv32/qemu_riscv32.dts [hunk 3]
```diff
@@ -1,23 +1,52 @@
 /dts-v1/;
 
-#include <riscv32-qemu.dtsi>
```
Comment (galak) [inline boards/riscv32/qemu_riscv32/qemu_riscv32.dts]: We should probably remove the file 'riscv32-qemu.dtsi'

**Review by galak — COMMENTED** [review]
Probably should remove UART_RISCV_QEMU related bits

File: boards/riscv32/qemu_riscv32/qemu_riscv32.dts [hunk 4]
```diff
@@ -1,23 +1,52 @@
...
 
 / {
-	model = "qemu_riscv32";
-	compatible = "qemu,riscv32";
-
+	model = "SiFive HiFive 1";
+	compatible = "sifive,hifive1";
 	aliases {
-		uart-0 = &uart0;
+		led0 = &led0;
```
Comment (galak) [inline boards/riscv32/qemu_riscv32/qemu_riscv32.dts]: are there really led's in the qemu model?  Seems like we shouldn't have them in the dts file.

**Review by galak — DISMISSED** [review]
I think we should replace qemu_cortex_m3 with mps2_an385 in the samples/tests.  We should have at least one board/platform on ARM for all those tests.

**Comment by nashif** [comment]
> I think we should replace qemu_cortex_m3 with mps2_an385 in the samples/tests. We should have at least one board/platform on ARM for all those tests.

ok, I just did a straight substitution now, will need to go cleanup for sanitycheck later.

**Comment by nashif** [comment]
@galak ok, I think we are close.

**Review by galak — CHANGES_REQUESTED** [review]
If we are going to remove the qemu_arm_board, we should probably remove the SoC support as well:

soc/arm/ti_lm3s6965/Kconfig:if SOC_TI_LM3S6965
soc/arm/ti_lm3s6965/Kconfig:config SOC_TI_LM3S6965_QEMU
soc/arm/ti_lm3s6965/Kconfig:endif # SOC_TI_LM3S6965
soc/arm/ti_lm3s6965/Kconfig.defconfig:if SOC_TI_LM3S6965
soc/arm/ti_lm3s6965/Kconfig.soc:config SOC_TI_LM3S6965
tests/kernel/arm_irq_vector_table/src/arm_irq_vector_table.c:#if defined(CONFIG_SOC_TI_LM3S6965_QEMU)
tests/kernel/gen_isr_table/src/main.c:#if defined(CONFIG_SOC_TI_LM3S6965_QEMU)
tests/kernel/interrupt/src/interrupt.h:#if defined(CONFIG_SOC_TI_LM3S6965_QEMU) || defined(CONFIG_CPU_CORTEX_M0) \
tests/kernel/tickless/tickless/src/main.c:#  if defined(CONFIG_SOC_TI_LM3S6965_QEMU)
tests/kernel/tickless/tickless/src/timestamps.c:#if defined(CONFIG_SOC_TI_LM3S6965_QEMU)

**Review by pfalcon — DISMISSED** [review]
> If we are going to remove the qemu_arm_board

I'm not sure what happens here. How can we remove any qemu board - there're not so many of them, and all are needed and helpful with testing!

**Comment by nashif** [comment]
> I and other people use qemu_cortex_m3 daily. It's developed further by the community: #10097 . I'm -1 on the change of removal qemu_cortex_m3.

Again, qemu_cortex_m3 will be covered EXACTLY the same way with mps2_an385, you can run it in qemu and doc exactly the same things you do right now with the additional benefit of having the new configuration support MPU as well.

**Comment by pfalcon** [comment]
> Again, qemu_cortex_m3 will be covered EXACTLY the same way with mps2_an385

Well, it won't be the same way. qemu_cortex_m3 and mps2_an385 are two different machine types in qemu, with different devices hardcoded to them, as qemu is nowhere near as flexible as to construct arbitrary machine with arbitrary devices. Specific examples: qemu_cortex_m3 has an Ethernet controller, for which the driver is already written: https://github.com/zephyrproject-rtos/zephyr/pull/10097 . For mps2_an385, the driver is yet to written, and well, it's *different* hardware, and different hardware never works exactly the same, even if emulated. Next, qemu_cortex_m3 works with SLIP networking. There're issues, *known* issues. Nothing is known whether mps2_an385 works with SLIP, or yet work needs to be done to enable that, and which issue it will have with it. Etc, etc.

Again, there can't be enough emulated boards, and killing 1 of the 2 available for ARM is hardly a wise move.

**Comment by nashif** [comment]
> Again, there can't be enough emulated boards, and killing 1 of the 2 available for ARM is hardly a wise move.


Ok, I will change this PR to leave qemu_cortex_m3 which emulates lm3s6965evb but will remove it as a default sanitycheck platform and use mps2_an385 instead. What will happen with lm3s6965evb emulation and if we want to keep it can be decided later.

**Comment by pfalcon** [comment]
> Ok, I will change this PR to leave qemu_cortex_m3 which emulates lm3s6965evb but will remove it as a default sanitycheck platform and use mps2_an385 instead. 

Sounds good, thanks.

> What will happen with lm3s6965evb emulation and if we want to keep it can be decided later.

We definitely can rename it from qemu_cortex_m3 into lm3s6965evb some time later.

**Review by galak — APPROVED** [review]

**Comment by galak** [comment]
We should send an email to the devel list about the update to 0.9.5

**Review by galak — DISMISSED** [review]
Need to remove uart_riscv_qemu.c from drivers/serial/CMakeLists.txt  and remove the file itself.

**Comment by galak** [comment]
Need to fix soc/riscv32/riscv-privilege/common/CMakeLists.txt reference to CONFIG_SOC_SERIES_RISCV32_QEMU

**Review by SebastianBoe — APPROVED** [review]

**Comment by nashif** [comment]
> Need to fix soc/riscv32/riscv-privilege/common/CMakeLists.txt reference to CONFIG_SOC_SERIES_RISCV32_QEMU

done

**Review by galak — APPROVED** [review]

**Comment by galak** [comment]
Still think we should email the devel list before merging this.

**Comment by nashif** [comment]
> Still think we should email the devel list before merging this.

yes, I have the email ready.

**Pull request merged**